http/http_client.hpp

```diff
@@ -497,14 +498,19 @@ class ConnectionInfo : public std::enable_shared_from_this<ConnectionInfo>
         }
 
         // Let's close the connection and restart from resolve.
-        doClose(true);
+        doCleanup(true);
```

Comment:
yes, we can do it. But giving explicit name similar to close looked readable. Also I felt it is easier to add new code (in case needed in future) before calling the shutdown. I am open to remove this function.